public/app/features/alerting/components/AlertingQueryEditor.tsx

```diff
@@ -134,26 +178,33 @@ const addQuery = (
     model: {
       ...queryToAdd.model,
       hide: false,
-      refId: refId,
-    },
-    relativeTimeRange: {
-      from: 21600,
-      to: 0,
+      refId,
     },
+    relativeTimeRange: defaultTimeRange(queryToAdd.model),
   };
 
   return [...queries, query];
 };
 
+const defaultTimeRange = (model: DataQuery): RelativeTimeRange | undefined => {
+  if (isExpressionQuery(model)) {
+    return;
+  }
+
+  return {
+    from: 21600,
+    to: 0,
+  };
+};
+
```

Comment:
We already have an getDefaultTimeRange function maybe we could call within this function?

Reply:
yeah, or I think we should add a `getDefaultRelativeTimeRange` along side that one.